src/arrowc/arrowc.h

```diff
@@ -203,6 +270,102 @@ ArrowErrorCode ArrowSchemaSetMetadata(struct ArrowSchema* schema, const char* me
 
 /// }@
 
+/// \defgroup arrowc-schema-view Schema consumer helpers
+
+/// \brief A non-owning view of a parsed ArrowSchema
+///
+/// Contains more readily extractable values than a raw ArrowSchema.
+/// Clients can stack or statically allocate this structure but are
+/// encouraged to use the provided getters to ensure forward
+/// compatiblity.
+struct ArrowSchemaView {
+  /// A pointer to the schema represented by this view
+  struct ArrowSchema* schema;
+
+  /// \brief The data type represented by the schema
+  ///
+  /// This value may be ARROWC_TYPE_DICTIONARY if the schema has a
+  /// non-null dictionary member; this value may be ARROWC_TYPE_EXTENSION
+  /// if the schema has the ARROW:extension:name metadata field set.
+  /// Datetime types are valid values.
+  enum ArrowType data_type;
+
+  /// \brief The storage data type represented by the schema
+  ///
+  /// This value will never be ARROWC_TYPE_DICTIONARY, ARROWC_TYPE_EXTENSION
+  /// or any datetime type. This value represents only the type required to
+  /// interpret the buffers in the array.
+  enum ArrowType storage_data_type;
+
+  /// \brief The expected number of buffers in a paired ArrowArray
+  int32_t n_buffers;
+
+  /// \brief The index of the validity buffer or -1 if one does not exist
+  int32_t validity_buffer_id;
+
+  /// \brief The index of the offset buffer or -1 if one does not exist
+  int32_t offset_buffer_id;
+
+  /// \brief The index of the large_offset buffer or -1 if one does not exist
+  int32_t large_offset_buffer_id;
```

Comment:
is it useful to differentiate between offset/large offset here? since you'd be looking at the type anyways